### ISSUE TITLE
Support local CSV files

### ISSUE THREAD

**Issue by marcusolsson** [grafana/grafana-csv-datasource#6]
Currently, the plugin only supports remote CSV files that are available over HTTP. There's no support for loading CSV files from disk.

**Comment by cpmitch** [comment]
Yes, please support local CSV files, thank you.

**Comment by ouinouin** [comment]
Hi, it would definitely de an open door to ease something more "local" and end user friendly.

**Comment by WhiteCrow** [comment]
Yes, I want to it support local CSV files, too. Thank you!

**Comment by liuning0820** [comment]
Yes, please support local CSV files

**Comment by mikethebee** [comment]
Yes, I would be useful for me too. Thank you.

**Comment by dinesh3367** [comment]
It would be great to have support for local CSV files

**Comment by marcusolsson** [comment]
This is clearly something we should add :) Will get working on this ASAP

**Comment by rmaus** [comment]
... and another voice in the choir:  "Yes, please."  Thanks!

**Comment by marcusolsson** [comment]
I've made good progress on this. Should be able to publish a release later this week 🙏

**Comment by marcusolsson** [comment]
I've published [v0.2.0](https://grafana.com/grafana/plugins/marcusolsson-csv-datasource?version=0.2.0) which contains initial support for local CSV files. Please try it out, I'll leave this issue open a while longer in case there are any obvious issues.

**Comment by mikethebee** [comment]
Thx, just tried it and it worked fine. btw, I also tried adding a space at the end of each line and that didn't cause a problem using the worldmap plugin.

**Comment by rmaus** [comment]
Marcus,

Thanks much.  I have been able to successfully import a CSV file with the 0.2.0 plugin.

**_One minor issue to report_**:  On Windows 10, importing files with backslashes in the path fails, e.g. c:\temp\file.csv, whereas importing files with forward slashes succeeds, e.g. c:/temp/file.csv

Thanks again,
Rick

**Comment by marcusolsson** [comment]
@rmaus Thanks for the report! 🙏 I created a separate issue to track this.

**Comment by marcusolsson** [comment]
Published [v0.3.0](https://grafana.com/grafana/plugins/marcusolsson-csv-datasource?pg=plugins&plcmt=featured-undefined&version=0.3.0) that (hopefully) fixes the Windows path issue. I don't have access to a Windows machine, so would appreciate confirmation that it actually works 😅

**Comment by suntong** [comment]
> Published [v0.3.0](https://grafana.com/grafana/plugins/marcusolsson-csv-datasource?pg=plugins&plcmt=featured-undefined&version=0.3.0) that (hopefully) fixes the Windows path issue. I don't have access to a Windows machine, so would appreciate confirmation that it actually works 😅

How can I tell which version I'm using? 
Anyway, I believe I have  [v0.3.0](https://grafana.com/grafana/plugins/marcusolsson-csv-datasource?pg=plugins&plcmt=featured-undefined&version=0.3.0) as that's what I downloaded today, and with that, 

- importing files with backslashes in the path still fails. 
- importing files with forward slashes succeeds as before. 

UPDATE: confirmed -- I have v0.3.0, as what I downloaded today is `marcusolsson-csv-datasource-0.3.0.zip`.
